File: MyPortal.Core/Entities/School.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MyPortal.Core.Entities
{
    [Table("Schools")]
    public class School : Entity
    {
        public Guid AgencyId { get; set; }

        public Guid? LocalAuthorityId { get; set; }

        public int EstablishmentNumber { get; set; }

        [Required]
        [StringLength(128)]
        public string Urn { get; set; } = null!;

        [Required]
        [StringLength(128)]
        public string Uprn { get; set; } = null!;

        public Guid SchoolPhaseId { get; set; }

        public Guid SchoolTypeId { get; set; }

        public Guid GovernanceTypeId { get; set; }

        public Guid IntakeTypeId { get; set; }

        public Guid? HeadTeacherId { get; set; }

        public Guid? PayZoneId { get; set; }

        // UK Provider Reference Number (8 digits) — distinct from URN and the UPRN property reference.
        [StringLength(8)]
        public string? Ukprn { get; set; }

        // Statutory age range (lowest / highest national-curriculum ages admitted).
        public int? LowestAge { get; set; }

        public int? HighestAge { get; set; }

        public int? NetCapacity { get; set; }

        public DateTime? NetCapacityAssessmentDate { get; set; }

        // Special-school establishment facts (CBDS CS038 / CS077); gated by IsSpecialSchool.
        public bool IsSpecialSchool { get; set; }

        public Guid? SpecialSchoolOrganisationId { get; set; }

        public Guid? SpecialSchoolTypeId { get; set; }

        // Maximum boarding pupils (boarding establishments).
        public int? MaxBoarders { get; set; }

        // Direct establishment contact (the school currently inherits this from its Agency).
        [StringLength(30)]
        public string? Telephone { get; set; }

        [StringLength(256)]
        public string? Email { get; set; }

        public bool IsLocal { get; set; }

        public Agency? Agency { get; set; }
        public SchoolPhase? SchoolPhase { get; set; }
        public SchoolType? SchoolType { get; set; }
        public GovernanceType? GovernanceType { get; set; }
        public IntakeType? IntakeType { get; set; }
        public Person? HeadTeacher { get; set; }
        public LocalAuthority? LocalAuthority { get; set; }
        public PayZone? PayZone { get; set; }
        public SpecialSchoolOrganisation? SpecialSchoolOrganisation { get; set; }
        public SpecialSchoolType? SpecialSchoolType { get; set; }
    }
}

File: MyPortal.Frontend/myportal-spa/src/app/shared/types/school.ts
// Wire shapes for the school details edit page. Mirrors
// MyPortal.Contracts.Models.School.* on the backend.

export interface SchoolDetailsResponse {
  id: string;
  agencyId: string;
  name: string;
  website?: string | null;
  agencyTypeId: string;
  urn: string;
  uprn: string;
  establishmentNumber: number;
  localAuthorityId?: string | null;
  localAuthorityName?: string | null;
  schoolPhaseId: string;
  phase: string;
  schoolTypeId: string;
  type: string;
  governanceTypeId: string;
  intakeTypeId: string;
  headTeacherId?: string | null;
  headTeacherFullName?: string | null;
  ukprn?: string | null;
  payZoneId?: string | null;
  payZoneName?: string | null;
  lowestAge?: number | null;
  highestAge?: number | null;
  netCapacity?: number | null;
  netCapacityAssessmentDate?: string | null;
  isSpecialSchool: boolean;
  specialSchoolOrganisationId?: string | null;
  specialSchoolOrganisationName?: string | null;
  specialSchoolTypeId?: string | null;
  specialSchoolTypeName?: string | null;
  maxBoarders?: number | null;
  telephone?: string | null;
  email?: string | null;
  isLocal: boolean;
}

export interface SchoolUpsertRequest {
  name: string;
  website?: string | null;
  expectedVersion: number;
  localAuthorityId?: string | null;
  establishmentNumber: number;
  urn: string;
  uprn: string;
  schoolPhaseId: string;
  schoolTypeId: string;
  governanceTypeId: string;
  intakeTypeId: string;
  headTeacherId?: string | null;
  ukprn?: string | null;
  payZoneId?: string | null;
  lowestAge?: number | null;
  highestAge?: number | null;
  netCapacity?: number | null;
  netCapacityAssessmentDate?: string | null;
  isSpecialSchool: boolean;
  specialSchoolOrganisationId?: string | null;
  specialSchoolTypeId?: string | null;
  maxBoarders?: number | null;
  telephone?: string | null;
  email?: string | null;
}

export interface LocalAuthoritySummaryResponse {
  id: string;
  leaCode: number;
  name: string;
}
